torch/_inductor/compile_fx.py

```diff
@@ -1061,9 +1064,20 @@ def log_graph_runnable() -> str:
                             disable = f"{disable} Found from {stack_trace}\n"
                         V.graph.disable_cudagraphs_reason = disable
 
-                if V.aot_compilation is True:
-                    assert isinstance(compiled_fn, str)
-                    return CompiledAOTI(compiled_fn)
+                if isinstance(compiled_fn, (str, list)):
+                    # TODO: unconditionally packaging this is not always
+                    # desirable
+                    # TODO: probably should live in codecache, maybe...
+                    from torch._inductor.codecache import get_hash, get_path
+                    from torch._inductor.package import package_aoti
+
+                    key = get_hash(":".join(compiled_fn), "", "code")
+                    basename, subdir, path = get_path(key, "pt2", "")
+                    pathlib.Path(path).parent.mkdir(parents=True, exist_ok=True)
+                    output_code_log.debug("Package written to %s", path)
```

Comment:
TODO: Need to do some CAS refactoring to make this better. package_aoti isn't really cooperating either. This needs to be atomic, not enough guts exposed.